### DISCUSSION TITLE
[BR-CO-26]-In order for the buyer to automatically identify a supplier, the Seller identifier (BT-29), the Seller legal registration identifier (BT-30) and/or the Seller VAT identifier (BT-31) shall be present.

### DISCUSSION THREAD

**Discussion by wh1980** [stephanstapel/ZUGFeRD-csharp#521] [General]
Hi, I'am getting error messages in three different ZUGFeRD validators for the following code when I only pass TaxRegistrationSchemeID.FC. Why is that? Many companies just don't have a sales tax id but only a national tax id. How can I solve the problem?

```
' m_Descriptor.AddSellerTaxRegistration("DE4455667788", TaxRegistrationSchemeID.VA) ' Umsatzsteuer-ID
m_Descriptor.AddSellerTaxRegistration("12345/67890", TaxRegistrationSchemeID.FC) ' Steuernummer
```

`<?xml version="1.0" encoding="utf-8"?>
<rsm:CrossIndustryInvoice xmlns:a="urn:un:unece:uncefact:data:standard:QualifiedDataType:100" xmlns:rsm="urn:un:unece:uncefact:data:standard:CrossIndustryInvoice:100" xmlns:qdt="urn:un:unece:uncefact:data:standard:QualifiedDataType:100" xmlns:ram="urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:100" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:udt="urn:un:unece:uncefact:data:standard:UnqualifiedDataType:100">
  <rsm:ExchangedDocumentContext>
    <ram:GuidelineSpecifiedDocumentContextParameter>
      <ram:ID>urn:cen.eu:en16931:2017#conformant#urn:factur-x.eu:1p0:extended</ram:ID>
    </ram:GuidelineSpecifiedDocumentContextParameter>
  </rsm:ExchangedDocumentContext>
  <rsm:ExchangedDocument>
    <ram:ID>00010</ram:ID>
    <ram:TypeCode>380</ram:TypeCode>
    <ram:IssueDateTime>
      <udt:DateTimeString format="102">20241119</udt:DateTimeString>
    </ram:IssueDateTime>
    <ram:IncludedNote>
      <ram:Content>Geschäftsführer bzw. Inhaber: Max Mustermann</ram:Content>
      <ram:SubjectCode>REG</ram:SubjectCode>
    </ram:IncludedNote>
    <ram:IncludedNote>
      <ram:Content>Gemäß § 19 UStG (1) enthält der Rechnungsbetrag keine Umsatzsteuer.</ram:Content>
      <ram:SubjectCode>AAI</ram:SubjectCode>
    </ram:IncludedNote>
    <ram:IncludedNote>
      <ram:Content>Sie sind verpflichtet, die Rechnung zu Steuerzwecken zehn Jahre lang aufzubewahren.</ram:Content>
      <ram:SubjectCode>AAI</ram:SubjectCode>
    </ram:IncludedNote>
    <ram:IncludedNote>
      <ram:Content>Die aufgeführten Lieferungen bzw. Leistungen wurden ausgeführt von November 2024 bis Dezember 2024.</ram:Content>
      <ram:SubjectCode>AAI</ram:SubjectCode>
    </ram:IncludedNote>
    <ram:IncludedNote>
      <ram:Content>Datenschutzmitteilung: Die Verarbeitung der personenbezogenen Daten erfolgt im Einklang mit und nach den Grundsätzen der EU-DSGVO. Sie haben ein Recht auf unentgeltliche Auskunft über Ihre gespeicherten Daten sowie ggf. ein Recht auf Berichtigung, Sperrung oder Löschung dieser Daten. Wenn Sie weitere Fragen zur Erhebung, Verarbeitung oder Nutzung Ihrer personenbezogenen Daten haben, kontaktieren Sie uns bitte oder informieren Sie sich auf unserer Website.</ram:Content>
      <ram:SubjectCode>AAI</ram:SubjectCode>
    </ram:IncludedNote>
  </rsm:ExchangedDocument>
  <rsm:SupplyChainTradeTransaction>
    <ram:IncludedSupplyChainTradeLineItem>
      <ram:AssociatedDocumentLineDocument>
        <ram:LineID>1</ram:LineID>
      </ram:AssociatedDocumentLineDocument>
      <ram:SpecifiedTradeProduct>
        <ram:GlobalID schemeID="0160">Keine Angabe</ram:GlobalID>
        <ram:Name>Fliesenarbeiten im Wohnzimmer</ram:Name>
        <ram:Description>Alte Fliesen entfernen und neu fliesen.</ram:Description>
        <ram:ApplicableProductCharacteristic>
          <ram:Description>EAN / GTIN</ram:Description>
          <ram:Value>Keine Angabe</ram:Value>
        </ram:ApplicableProductCharacteristic>
        <ram:ApplicableProductCharacteristic>
          <ram:Description>Kategorie</ram:Description>
          <ram:Value>Fliesenarbeiten</ram:Value>
        </ram:ApplicableProductCharacteristic>
      </ram:SpecifiedTradeProduct>
      <ram:SpecifiedLineTradeAgreement>
        <ram:GrossPriceProductTradePrice>
          <ram:ChargeAmount>300.00</ram:ChargeAmount>
          <ram:BasisQuantity unitCode="LS">1.0000</ram:BasisQuantity>
        </ram:GrossPriceProductTradePrice>
        <ram:NetPriceProductTradePrice>
          <ram:ChargeAmount>300.00</ram:ChargeAmount>
          <ram:BasisQuantity unitCode="LS">1.0000</ram:BasisQuantity>
        </ram:NetPriceProductTradePrice>
      </ram:SpecifiedLineTradeAgreement>
      <ram:SpecifiedLineTradeDelivery>
        <ram:BilledQuantity unitCode="LS">1.0000</ram:BilledQuantity>
      </ram:SpecifiedLineTradeDelivery>
      <ram:SpecifiedLineTradeSettlement>
        <ram:ApplicableTradeTax>
          <ram:TypeCode>VAT</ram:TypeCode>
          <ram:ExemptionReason>steuerbefreit</ram:ExemptionReason>
          <ram:CategoryCode>E</ram:CategoryCode>
          <ram:RateApplicablePercent>0.00</ram:RateApplicablePercent>
        </ram:ApplicableTradeTax>
        <ram:SpecifiedTradeSettlementLineMonetarySummation>
          <ram:LineTotalAmount>300.00</ram:LineTotalAmount>
        </ram:SpecifiedTradeSettlementLineMonetarySummation>
      </ram:SpecifiedLineTradeSettlement>
    </ram:IncludedSupplyChainTradeLineItem>
    <ram:ApplicableHeaderTradeAgreement>
      <ram:SellerTradeParty>
        <ram:Name>Malerbetrieb Mustermann</ram:Name>
        <ram:DefinedTradeContact>
          <ram:PersonName>Max Mustermann</ram:PersonName>
          <ram:TelephoneUniversalCommunication>
            <ram:CompleteNumber>0172 / 123 456 789</ram:CompleteNumber>
          </ram:TelephoneUniversalCommunication>
          <ram:EmailURIUniversalCommunication>
            <ram:URIID>info@mustermann.de</ram:URIID>
          </ram:EmailURIUniversalCommunication>
        </ram:DefinedTradeContact>
        <ram:PostalTradeAddress>
          <ram:PostcodeCode>76133</ram:PostcodeCode>
          <ram:LineOne>Max Mustermann</ram:LineOne>
          <ram:LineTwo>Musterstr. 1</ram:LineTwo>
          <ram:CityName>Karlsruhe</ram:CityName>
          <ram:CountryID>DE</ram:CountryID>
        </ram:PostalTradeAddress>
        <ram:SpecifiedTaxRegistration>
          <ram:ID schemeID="FC">12345/67890</ram:ID>
        </ram:SpecifiedTaxRegistration>
      </ram:SellerTradeParty>
      <ram:BuyerTradeParty>
        <ram:ID>1</ram:ID>
        <ram:Name>Schuster Ltd.</ram:Name>
        <ram:DefinedTradeContact>
          <ram:PersonName>Winfried Schuster</ram:PersonName>
          <ram:TelephoneUniversalCommunication>
            <ram:CompleteNumber>0177 / 78 50 50</ram:CompleteNumber>
          </ram:TelephoneUniversalCommunication>
          <ram:EmailURIUniversalCommunication>
            <ram:URIID>winfried.schuster@online.de</ram:URIID>
          </ram:EmailURIUniversalCommunication>
        </ram:DefinedTradeContact>
        <ram:PostalTradeAddress>
          <ram:PostcodeCode>76133</ram:PostcodeCode>
          <ram:LineOne>Winfried Schuster</ram:LineOne>
          <ram:LineTwo>Hardtstr. 2</ram:LineTwo>
          <ram:CityName>Karlsruhe</ram:CityName>
          <ram:CountryID>DE</ram:CountryID>
        </ram:PostalTradeAddress>
        <ram:SpecifiedTaxRegistration>
          <ram:ID schemeID="VA">DE4455667788</ram:ID>
        </ram:SpecifiedTaxRegistration>
      </ram:BuyerTradeParty>
    </ram:ApplicableHeaderTradeAgreement>
    <ram:ApplicableHeaderTradeDelivery>
      <ram:ActualDeliverySupplyChainEvent>
        <ram:OccurrenceDateTime>
          <udt:DateTimeString format="102">20241231</udt:DateTimeString>
        </ram:OccurrenceDateTime>
      </ram:ActualDeliverySupplyChainEvent>
    </ram:ApplicableHeaderTradeDelivery>
    <ram:ApplicableHeaderTradeSettlement>
      <ram:PaymentReference>00010</ram:PaymentReference>
      <ram:InvoiceCurrencyCode>EUR</ram:InvoiceCurrencyCode>
      <ram:SpecifiedTradeSettlementPaymentMeans>
        <ram:TypeCode>58</ram:TypeCode>
        <ram:Information>SEPA-Überweisung</ram:Information>
        <ram:PayeePartyCreditorFinancialAccount>
          <ram:IBANID>DE89370400440532013000</ram:IBANID>
          <ram:AccountName>Malerbetrieb Mustermann</ram:AccountName>
        </ram:PayeePartyCreditorFinancialAccount>
        <ram:PayeeSpecifiedCreditorFinancialInstitution>
          <ram:BICID>COBADEFFXXX</ram:BICID>
        </ram:PayeeSpecifiedCreditorFinancialInstitution>
      </ram:SpecifiedTradeSettlementPaymentMeans>
      <ram:ApplicableTradeTax>
        <ram:CalculatedAmount>0.00</ram:CalculatedAmount>
        <ram:TypeCode>VAT</ram:TypeCode>
        <ram:ExemptionReason>Gemäß § 19 UStG (1) enthält der Rechnungsbetrag keine Umsatzsteuer.</ram:ExemptionReason>
        <ram:BasisAmount>300.00</ram:BasisAmount>
        <ram:CategoryCode>E</ram:CategoryCode>
        <ram:RateApplicablePercent>0.00</ram:RateApplicablePercent>
      </ram:ApplicableTradeTax>
      <ram:BillingSpecifiedPeriod>
        <ram:StartDateTime>
          <udt:DateTimeString format="102">20241101</udt:DateTimeString>
        </ram:StartDateTime>
        <ram:EndDateTime>
          <udt:DateTimeString format="102">20241231</udt:DateTimeString>
        </ram:EndDateTime>
      </ram:BillingSpecifiedPeriod>
      <ram:SpecifiedTradePaymentTerms>
        <ram:Description>Überweisen Sie bitte den offenen Betrag in Höhe von 300,00 € auf das unten aufgeführte Geschäftskonto.</ram:Description>
        <ram:DueDateDateTime>
          <udt:DateTimeString format="102">20250102</udt:DateTimeString>
        </ram:DueDateDateTime>
      </ram:SpecifiedTradePaymentTerms>
      <ram:SpecifiedTradeSettlementHeaderMonetarySummation>
        <ram:LineTotalAmount>300.00</ram:LineTotalAmount>
        <ram:ChargeTotalAmount>0.00</ram:ChargeTotalAmount>
        <ram:AllowanceTotalAmount>0.00</ram:AllowanceTotalAmount>
        <ram:TaxBasisTotalAmount>300.00</ram:TaxBasisTotalAmount>
        <ram:TaxTotalAmount currencyID="EUR">0.00</ram:TaxTotalAmount>
        <ram:RoundingAmount>0.00</ram:RoundingAmount>
        <ram:GrandTotalAmount>300.00</ram:GrandTotalAmount>
        <ram:TotalPrepaidAmount>0.00</ram:TotalPrepaidAmount>
        <ram:DuePayableAmount>300.00</ram:DuePayableAmount>
      </ram:SpecifiedTradeSettlementHeaderMonetarySummation>
    </ram:ApplicableHeaderTradeSettlement>
  </rsm:SupplyChainTradeTransaction>
</rsm:CrossIndustryInvoice>`

### Mustang and valitool
URL: [https://www.portinvoice.com](url)

Error
[BR-CO-26]-In order for the buyer to automatically identify a supplier, the Seller identifier (BT-29), the Seller legal registration identifier (BT-30) and/or the Seller VAT identifier (BT-31) shall be present. [ID FX-SCH-A-000001] from /xslt/ZF_232/FACTUR-X_EXTENDED.xslt)
/*:CrossIndustryInvoice[namespace-uri()='urn:un:unece:uncefact:data:standard:CrossIndustryInvoice:100'][1]/*:SupplyChainTradeTransaction[namespace-uri()='urn:un:unece:uncefact:data:standard:CrossIndustryInvoice:100'][1]/*:ApplicableHeaderTradeAgreement[namespace-uri()='urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:100'][1]/*:SellerTradeParty[namespace-uri()='urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:100'][1]

Details
[BR-CO-26]-Damit der Erwerber den Lieferanten automatisch identifizieren kann, soll der „Seller identifier“ (BT-29), der „Seller legal registration identifier“ (BT-30) oder der „Seller VAT identifier“ (BT-31) vorhanden sein.

### Invoice-Portal Validator
URL: https://validator.invoice-portal.de

Schematron (XSLT)
CEN-EN16931-CII
1
Fatal
[BR-CO-26]-In order for the buyer to automatically identify a supplier, the Seller identifier (BT-29), the Seller legal registration identifier (BT-30) and/or the Seller VAT identifier (BT-31) shall be present.
Schematron (XSLT)
ZUGFeRD 2.2 Extended

1
In order for the buyer to automatically identify a supplier, the Seller identifier (BT-29), the Seller legal registration identifier (BT-30) and/or the Seller VAT identifier (BT-31) shall be present.

**Comment by stephanstapel** [comment]
BT-30 hast Du eingetragen?

**Reply by wh1980** [reply]
Wo trage ich es ein? Mit welcher Funktion?
Danke übrigens für Dein Framework

**Reply by stephanstapel** [reply]
![grafik](https://github.com/user-attachments/assets/901692aa-ab05-4f28-a858-eb1684a2368f)

also entsprechend legalOrganization hier füllen:

![grafik](https://github.com/user-attachments/assets/b97455ce-6385-49f9-a2b2-8676d31b077a)

**Reply by wh1980** [reply]
Das hatte ich auch versucht, nur konnte ich keinen passenden Code in GlobalIDSchemeIdentifiers finden. Wäre GlobalIDSchemeIdentifiers.CompanyNumber die richtige Konstante für die nationale Steuernummer? Jedenfalls spucken die Validatoren nun keine Fehler mehr aus. Vielen Dank.

`seller.SpecifiedLegalOrganization = New LegalOrganization(GlobalIDSchemeIdentifiers.CompanyNumber, "12345/67890")}`

Und weiter zwei Fragen zum Verständnis:

Der generierte Code enthält nun **ram:SpecifiedLegalOrganization**.
Weiter unten habe ich aber noch `<ram:ID schemeID="FC">12345/67890</ram:ID>` stehen.
Für was wäre diese Angabe, wenn bereits `<ram:SpecifiedLegalOrganization>` gesetzt ist?
Und zur Umsatzsteuer-ID: `<ram:ID schemeID="VA">DE123456789</ram:ID>` wäre an dieser Stelle richtig, nehme ich an?

```
    <ram:SellerTradeParty>
       ...
        <ram:SpecifiedLegalOrganization>
          <ram:ID schemeID="0208">12345/67890</ram:ID>
        </ram:SpecifiedLegalOrganization>
        ...
        <ram:SpecifiedTaxRegistration>
          <ram:ID schemeID="VA">DE123456789</ram:ID>
        </ram:SpecifiedTaxRegistration>
        <ram:SpecifiedTaxRegistration>
          <ram:ID schemeID="FC">12345/67890</ram:ID>
        </ram:SpecifiedTaxRegistration>
      </ram:SellerTradeParty>
```

**Reply by stephanstapel** [reply]
in der Liste:

https://github.com/stephanstapel/ZUGFeRD-csharp/blob/3e8a706f67fff8e0551f363394a1fa522f9ba181/documentation/zugferd23de/Dokumentation/4.%20EN16931%2BFacturX%20code%20lists%20values%20v14%20-%20used%20from%202024-11-15.xlsx

sind noch einige anderen Einträge vorhanden, wie womöglich passen köntnen. Ich weigere mich lediglich, pauschal alle ISO-Listen vorzusehen, da dann niemand mehr irgendwas findet, vor lauter 'Amazon Unique Identification Scheme ', 'Vodafone Ireland OSI Addressing' und anderen sehr spezifischen Identifiern

Zu den weiteren Fragen: schau am besten in der technischen Dokumentation:

https://github.com/stephanstapel/ZUGFeRD-csharp/blob/3e8a706f67fff8e0551f363394a1fa522f9ba181/documentation/zugferd23de/Dokumentation/8.%20Factur-X_1.0.07_Technischer-Anhang_Profil_EXTENDED.pdf

Die Dokumentation ist kaum lesbar, aber man kann gut Einträge per Suche finden. Dort sind alle BT-* Referenzen vorhanden.

Und die Umsatzsteuer-ID sieht so korrekt aus, ja.

**Reply by wh1980** [reply]
Vielen Dank für Deine Antwort.
